### PR TITLE
8314094: java/lang/ProcessHandle/InfoTest.java fails on Windows when run as user with Administrator privileges

### PR DESCRIPTION
**Pull request by RealCLanger**
On Windows, the test java/lang/ProcessHandle/InfoTest.java can fail when run as user that is member of the Administrators group. In that case new files are not owned by the user but instead by BUILTIN\ADMINISTRATORS. This breaks the assumptions of the test's whoami check. My suggestion is to cater for this case and don't fail the test but write a warning message to stdout that a whoami check is not correctly possible.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8314094](https://bugs.openjdk.org/browse/JDK-8314094): java/lang/ProcessHandle/InfoTest.java fails on Windows when run as user with Administrator privileges (**Bug** - P4)

### Reviewers
 * [Matthias Baesken](https://openjdk.org/census#mbaesken) (@MBaesken - **Reviewer**) ⚠️ Review applies to [4e5cbf82](https://git.openjdk.org/jdk/pull/15222/files/4e5cbf82525e0344e77777fdf3c834b585d6bd15)
 * [Arno Zeller](https://openjdk.org/census#azeller) (@ArnoZeller - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15222/head:pull/15222` \
`$ git checkout pull/15222`

Update a local copy of the PR: \
`$ git checkout pull/15222` \
`$ git pull https://git.openjdk.org/jdk.git pull/15222/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15222`

View PR using the GUI difftool: \
`$ git pr show -t 15222`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15222.diff">https://git.openjdk.org/jdk/pull/15222.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15222#issuecomment-1672959991)

### CODE REVIEW

**Review by MBaesken — APPROVED** [review]
Looks simple and solves our issue.

File: test/jdk/java/lang/ProcessHandle/InfoTest.java [hunk 1]
```diff
@@ -294,7 +299,12 @@ public static void test3() {
                 if (info.user().isPresent()) {
                     String user = info.user().get();
                     Assert.assertNotNull(user);
-                    Assert.assertEquals(user, whoami);
+                    if (Platform.isWindows() && "BUILTIN\\Administrators".equals(whoami)) {
+                        System.out.println("Test seems to be run as Administrator. " +
+                                "Check for user correctness is not possible.");
```
[the remark this answers is not in the document]
Comment (ArnoZeller) [reply test/jdk/java/lang/ProcessHandle/InfoTest.java:304]: I think you might use System.getProperty("user.name"). But I am not sure about domain names of users on Windows.
I am also not sure why the user name is currently determined by creating a file - there might be a reason for this that is not obvious to me.

[the remark this answers is not in the document]
Comment (ArnoZeller) [reply test/jdk/java/lang/ProcessHandle/InfoTest.java:304]: It seems that ProcessHandle.info()  returns **DOMAIN/USERNAME** on Windows but System.getProperty("user.name") only the **USERNAME**.
You can get **DOMAIN** and **USERNAME** on **Windows** by calling:
com.sun.security.auth.module.NTSystem NTSystem = new com.sun.security.auth.module.NTSystem();
String user = NTSystem.getName();
String domain = NTSystem.getDomain();

[the remark this answers is not in the document]
Comment (RogerRiggs) [reply test/jdk/java/lang/ProcessHandle/InfoTest.java:304]: ok, more overhead than value in the suggestion. 
If its windows, strip off the domain (before "/") and compare.
If that doesn't work then just drop the username compare on windows.

[the remark this answers is not in the document]
Comment (RealCLanger) [reply test/jdk/java/lang/ProcessHandle/InfoTest.java]: I updated the change to use System.getProperty("user.name") in the Windows Administrators case. Let's see how testing goes.

**Comment by RealCLanger** [comment]
OK, the latest update seems to work. I'll integrate tomorrow, unless I hear concerns.

**Review by ArnoZeller — APPROVED** [review]
Looks good to me.

**Review by RogerRiggs — COMMENTED** [review]
A bit late due to a US holiday. Looks good.

**Comment by RealCLanger** [comment]
> A bit late due to a US holiday. Looks good.

Thanks 🙇

**Pull request closed without merging**